CONTRIBUTING.md

```diff
@@ -50,6 +50,8 @@ like this:
 ```
 python3 runtests.py
 ```
+*Please note that the python version used here is ``3.7``. This is due to the fact that we want to always type check in 3.7 mode to make sure we don't use any features from more recent Python versions.*
+*You can change this in the ``mypy_self_check.ini`` file.*
```

Comment:
I'm not sure if we should change `CONTRIBUTING.md`. If the dependencies are set up correctly, things should just work, and these additions are unnecessary detail. I don't think that changing the version is part of a normal workflow -- it should only be needed when dropping support for a Python version that has reached its end of life, and that's not something we need to document here.